client/src/components/side-bar.tsx

```diff
@@ -0,0 +1,614 @@
+  ContentCopy as ContentCopyIcon,
+} from "@mui/icons-material";
+import MuiDrawer from "@mui/material/Drawer";
+
+import { Lesson, LessonExpectation, TrainState } from "types";
+import { validateExpectationFeatures } from "schemas/validation";
+import SessionContext from "context/session";
+import { ToastContainer, toast } from "react-toastify";
+import { navigate } from "gatsby";
+import { OPENAI_CLASSIFIER_ARCHITECTURE } from "admin-constants";
+import { fetchLesson, updateLesson } from "api";
+import { useWithTraining } from "hooks/use-with-training";
+declare module "@mui/material/styles" {
+  interface Palette {
+    primary: Palette["primary"];
+    secondary: Palette["primary"];
+    error: Palette["primary"];
+    warning: Palette["primary"];
+    success: Palette["primary"];
+  }
+
+  interface PaletteOptions {
+    primary?: PaletteOptions["primary"];
+    secondary?: PaletteOptions["primary"];
+    error?: PaletteOptions["primary"];
+    warning?: PaletteOptions["primary"];
+    success?: PaletteOptions["primary"];
+  }
+}
+
+const buttonTheme = createTheme({
+  palette: {
+    primary: {
+      main: "#0C60AD",
+    },
+    secondary: {
+      main: "#000000",
+    },
+    warning: {
+      main: "#FFFF00",
+      contrastText: "#000000",
+    },
+    success: {
+      main: "#008000",
+    },
+    error: {
+      main: "#FF0000",
+    },
+  },
+});
+
+interface LessonUnderEdit {
+  lesson?: Lesson;
+  dirty?: boolean;
+}
+
+interface SideBarClasses {
+  drawer: string;
+  actionFooter: string;
+}
+export interface LessonEditSearch {
+  lessonId: string;
+  trainStatusPollInterval?: number;
+  copyLesson?: string;
+}
+
+export function SideBar(props: {
+  lessonId: string | "";
+  lessonUnderEdit: LessonUnderEdit;
+  error: string | "";
+  cookies: { [key: string]: string };
+  setLesson: (lesson?: Lesson, dirty?: boolean) => void;
+  setLessonId: (value: string) => void;
+  search: LessonEditSearch;
+  classes: SideBarClasses;
+}): JSX.Element {
+  const {
+    lessonId,
+    lessonUnderEdit,
+    error,
+    cookies,
+    setLesson,
+    setLessonId,
+    classes,
+  } = props;
+  const [savePopUp, setSavePopUp] = React.useState(false);
+
+  const context = useContext(SessionContext);
+  const {
+    isDownloadable,
+    isDownloading,
+    downloadMessage,
+    download,
+    dismissDownloadMessage,
+  } = useWithDownload(lessonId, context.user, cookies.accessToken);
+
+  const {
+    isTraining,
+    trainStatus,
+    trainingMessage,
+    startLessonTraining,
+    dismissTrainingMessage,
+  } = useWithTraining(props.search.trainStatusPollInterval);
+
+  React.useEffect(() => {
+    if (trainStatus.state === TrainState.SUCCESS) {
+      fetchLesson(lessonId ?? "", cookies.accessToken)
+        .then((lesson) => {
+          setLesson(lesson);
+        })
+        .catch((err) => console.error(err));
+    }
+  }, [trainStatus]);
+
+  const drawerWidth = 240;
+  const openedMixin = (theme: Theme): CSSObject => ({
```

Comment:
If possible, I'd move these CSSObjects and styled components to a separate file to clean up this component. This component is getting very bloated.